### PR TITLE
Use upstream patterns for DecomposeLinalgGeneric.

### PR DESCRIPTION
**Pull request by stellaraccident**

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir [hunk 1]
```diff
@@ -1,81 +1,21 @@
...
+  // CHECK: linalg.generic
+  // CHECK: arith.addf
+  // CHECK: linalg.generic
+  // CHECK: arith.mulf
   %12 = linalg.generic {
       indexing_maps = [affine_map<(d0, d1) -> (d1)>, affine_map<(d0, d1) -> (d0, d1)>],
       iterator_types = ["parallel", "parallel"]}
       ins(%arg0 : tensor<64xf32>) outs(%arg1 : tensor<?x64xf32>) {
   ^bb0(%arg2: f32, %arg3: f32):
-    %13 = arith.addf %arg2, %arg3 : f32
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir:78]: The semantics of reading the argument for the outs when everything is parallel is not clear to me (if you are read-modify-write-ing you should have a "reduction" iterator type). Thats why the upstream probably doesnt handle this. If needed I can fix that.

**Review by MaheshRavishankar — APPROVED** [review]
Thanks!

File: compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir (hunk 1 above)
Comment (stellaraccident) [reply compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir:78]: I get something like this with "bias adds":

```
    linalg.fill ins(%cst : f32) outs(%3 : memref<2x1xf32>)
    linalg.matmul {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[2, 0, 0]]>} ins(%0, %1 : memref<2x3xf32>, memref<3x1xf32>) outs(%3 : memref<2x1xf32>)
    linalg.generic {indexing_maps = [affine_map<(d0, d1) -> ()>, affine_map<(d0, d1) -> (d0, d1)>], iterator_types = ["parallel", "parallel"]} ins(%2 : memref<f32>) outs(%3 : memref<2x1xf32>) {                                ^bb0(%arg1: f32, %arg2: f32):                                                                                                                                                                                                  %6 = arith.addf %arg1, %arg2 : f32                                                                                                                                                                                           linalg.yield %6 : f32
    }
```

But I don't have to deal with them right now. And there may be something not right at the source.

Comment (stellaraccident) [reply compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir:78]: (these come from a tosa fully_connected)

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir:78]: ok, I see why this is happening. This is an artifact of having destination passing style and getting the computation bufferized in place without allocating scratch memory. I'll fix this issue upstream.

**Pull request merged**